xbmc/addons/AddonManager.cpp

```diff
@@ -1088,6 +1088,16 @@ bool CAddonMgr::IsFromOfficialRepo(const AddonPtr& addon, bool bCheckAddonPath)
          std::any_of(m_officialAddonRepos.begin(), m_officialAddonRepos.end(), comparator);
 }
 
+AddonOriginType CAddonMgr::GetAddonOriginType(const AddonPtr& addon) const
```

Comment:
Shouldn't this be in addontypes rather than in the main add-on file/definition?

Reply:
It requires the origin so I felt it fit better in the main class. It’s the type of the origin and not the addon itself.